### PR TITLE
FEAT: `np.tri`

### PR DESCRIPTION
**Pull request by aa452948257**
## What do these changes do?
Add function `np.tri`

Resolves: #277 

- [x] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/_mars/tensor/datasource/tri.py [hunk 1]
```diff
@@ -212,3 +213,50 @@ def tril(m, k=0, gpu=None):
...
+    array([[0.,  0.,  0.,  0.,  0.],
+           [1.,  0.,  0.,  0.,  0.],
+           [1.,  1.,  0.,  0.,  0.]])
+    """
+    if M is None:
+        M = N
+    m = asones((N, M), dtype=dtype, chunk_size=chunk_size, gpu=gpu)
+    gpu = m.op.gpu if gpu is None else gpu
+    op = TensorTril(k, dtype=m.dtype, sparse=m.issparse(), gpu=gpu)
+    return op(m)
```
Comment (UranusSeven) [inline python/xorbits/_mars/tensor/datasource/tri.py:262]: seems this impl equals `return m.tril(k)`?

**Review by UranusSeven — APPROVED** [review]
LGTM

File: python/xorbits/_mars/tensor/datasource/tests/test_datasource_execution.py [hunk 2]
```diff
@@ -822,6 +823,37 @@ def test_triu_execution(setup):
     assert res.flags["F_CONTIGUOUS"] == expected.flags["F_CONTIGUOUS"]
 
 
+def test_tri_execution(setup):
```
Comment (aresnow1) [inline python/xorbits/_mars/tensor/datasource/tests/test_datasource_execution.py:826]: Could you add some tests for `chunk_size` parameter?

Comment (aa452948257) [reply python/xorbits/_mars/tensor/datasource/tests/test_datasource_execution.py:826]: Added.

**Review by aresnow1 — APPROVED** [review]
LGTM

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

**Pull request merged**